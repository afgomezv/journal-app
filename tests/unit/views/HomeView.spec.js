import { shallowMount } from "@vue/test-utils";
import HomeView from "@/views/HomeView";

describe("AboutView Component", () => {
  test("debe de hacer match con el snapshot", () => {
    const wrapper = shallowMount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("hacer click en un boton deb redireccionar a no-entry", () => {
    const mockRouter = {
      push: jest.fn(),
    };

    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
