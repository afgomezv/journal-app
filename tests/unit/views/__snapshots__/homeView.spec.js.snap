// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView Component debe de hacer match con el snapshot 1`] = `
<div><img alt="Vue logo" src="">
  <h1>Hola Mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Succes</button>
</div>
`;
